src/dll/Systems/ScriptSystem.hpp

```diff
@@ -0,0 +1,37 @@
+#pragma once
+
+#include "Hook.hpp"
+#include "ISystem.hpp"
+#include "Paths.hpp"
+#include "PluginBase.hpp"
+
+class ScriptSystem : public ISystem
+{
+public:
+    ScriptSystem(const Paths& aPaths);
+
+    ESystemType GetType() final;
+
+    void Startup() final;
+    void Shutdown() final;
+
+    bool Add(std::shared_ptr<PluginBase> aPlugin, const char* path);
```

Comment:
```suggestion
    bool AddPath(std::shared_ptr<PluginBase> aPlugin, const char* path);
```

Reply:
I don't mind either way, but I was trying to match the existing pattern (verb w/o noun) used in the other systems, like `StateSystem::Add` & `HookingSystem::Attach`.

Reply:
It is mostly related to https://github.com/WopsS/RED4ext/pull/21#discussion_r1181266654.

Reply:
With the system rename, is the double `Add` ok here?

Reply:
I would say yes, at least for now.